_posts/2025-04-22-test-classloading-rewrite.adoc

```diff
@@ -0,0 +1,63 @@
+---
+layout: post
+title: 'The internals (and a few externals) of Quarkus test classloading have changed'
+tags: announcement testing
+synopsis: 'The way that Quarkus loads test classes has been updated. Most tests will not need to change, but here are some things to watch out for.'
+author: hcummins
+---
+
+== What's changing?
+
+The internals of Quarkus test classloading have been rewritten in 3.22.
+In particular, how `@QuarkusTest` test classes are loaded is changing.
+Other test extensions haven't changed, nor has dev mode and production mode classloading.
```

Comment:
If we say it before, we can remove this sentence.

Reply:
@maxandersen suggested the sentence, to make it very clear to people who might be skimming that production mode wasn't changing.

Reply:
I will go with the last one.